linter.py

```diff
@@ -13,52 +13,40 @@
 import subprocess
 import os
 import re
-import tempfile
 import time
+import socket
 
 from SublimeLinter.lint import Linter, util, persist
 
 
-def open_tmpfile(filename, suffix, code):
-    """Open and return a tempfile with the code in it. Remember to close this file."""
-    if not filename:
-        basename = util.UNSAVED_FILENAME
-        dirname = util.tempdir
-    else:
-        dirname, basename = os.path.split(filename)
-
-    if suffix:
-        basename = os.path.splitext(basename)[0] + suffix
-
-    f = tempfile.NamedTemporaryFile(dir=dirname, suffix=basename)
-    try:
-        if isinstance(code, str):
-            code = code.encode('utf-8')
-
-            f.write(code)
-            f.flush()
-    except:
-        f.close()
-
-    return f
-
-
-def call_server(message, address, port, timeout=2, sleep=2):
+def call_server(path, code, address, port, timeout=60):
     """Pass message to a juliaLint server at given address and port and return the response."""
-    command = ('(echo {}; sleep {}) | nc -w {} {} {}'.format(message, sleep, timeout, address, port))
-    return subprocess.check_output(command, shell=True).decode("utf-8")
-
-
-def launch_local_server(port):
-    """launch a julia lintserver."""
-    serverscript = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'lint/runjuliaserver.py')
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(timeout)
+    s.connect((address, port))
+    s.sendall(bytes(path, 'UTF-8'))
+    s.sendall(b"\n")
+    s.sendall(bytes(str(len(code)), 'UTF-8'))
+    s.sendall(b"\n")
+    s.sendall(bytes(code, 'UTF-8'))
+    s.sendall(b"\n")
+    response = s.recv(4096).decode("utf-8")
+    while response[-2:] not in ('\n\n', '\n'):
```

Comment:
An empty line is the signal for no more messages. I look at the last 2 characters so the only time the single new line can happen is if it's the first char sent, which is when the first line is empty. i.e there are no messages.